sql/types/number.go

```diff
@@ -806,6 +806,10 @@ func convertToInt64(t NumberTypeImpl_, v interface{}) (int64, sql.ConvertInRange
 		}
 		return i, sql.InRange, nil
 	case string:
+		if v == "" {
+			// StringType{}.Zero() returns empty string, but should represent "0" for number value
+			return 0, sql.InRange, nil
```

Comment:
interesting! I was kinda surprised that in MySQL `0 = ""` evaluates to true, but it totally does. Nice catch.